### PR TITLE
Ensure everything is using the same type structure

### PR DESCRIPTION
**Pull request by jason0x43**
There's much less string processing, and now (hopefully) all the plugins are using the same type / value structure. I also got to use `satisfies` for something! 😀

### CODE REVIEW

File: packages/glsl-ts-util/src/index.ts [hunk 1]
```diff
@@ -30,85 +72,88 @@ type IdentifierInitializer = {
...
-    } else {
-      sizes.push(0);
-      tokens.shift();
-    }
-  }
+// From https://registry.khronos.org/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf
+// GLSL 1 varyings are GLSL 3 vertex shader outputs / fragment shader inputs.
+// Fragment shader inputs cannot be booleans, opaque types, arrays of arrays or
+// structs, or structs containing arrays or structs.
+export type Varying = GlslField | GlslField[] | Record<string, GlslField>;
```
Comment (rorticus) [inline packages/glsl-ts-util/src/index.ts:115]: Hmm, either the comment is wrong or this type is wrong?

File: packages/glsl-ts-util/src/index.ts [hunk 2]
```diff
@@ -318,10 +366,10 @@ function isStructDeclaration(node: DeclarationStatementNode) {
  * Create a visitor that visits only root level declarations
  */
 function createRootDeclarationVisitor(
-  uniforms: Record<string, InternalShaderValue>,
-  attributes: Record<string, InternalShaderValue>,
-  varyings: Record<string, InternalShaderValue>,
-  structs: Record<string, Struct>,
+  uniforms: Record<string, Uniform>,
```
Comment (rorticus) [inline packages/glsl-ts-util/src/index.ts:369]: Can this be `Uniforms`, `Attributes`, and `Varyings` types now?

**Review by rorticus — APPROVED** [review]
I love it!

File: packages/glsl-ts-util/src/index.ts (hunk 2 above)
Comment (jason0x43) [reply packages/glsl-ts-util/src/index.ts:369]: Yup. I thought I had updated all of those, but apparently not. 🙂

File: packages/glsl-ts-util/src/index.ts (hunk 1 above)
Comment (jason0x43) [reply packages/glsl-ts-util/src/index.ts:115]: Varyings an be arrays and structs, but only of simple types.

Comment (jason0x43) [reply packages/glsl-ts-util/src/index.ts:115]: Oh, yeah, true. Attribute is wrong as far as that goes, too. I'll update it.

**Pull request merged**